BT Audio PACS code fails to build in regression tests with gcc12

**Describe the bug**
BT Audio PACS code fails to build in regression tests with gcc12

**To Reproduce**
Steps to reproduce the behavior:
1. cmake -GNinja -DBOARD=nrf52_bsim -DCMAKE_C_FLAGS="-Werror" <path>/zephyr/tests/bsim/bluetooth/audio && ninja 

**Expected behavior**
No build warnings, and therefore no regression tests errors

**Impact**
Blocks development with gcc 12 (unless BT audio regression tests are skipped)

**Logs and console output**
```
zephyr/subsys/bluetooth/audio/pacs.c
zephyr/subsys/bluetooth/audio/pacs.c: In function ‘pac_notify_loc’:
zephyr/subsys/bluetooth/audio/pacs.c:662:15: error: dangling pointer ‘uuid’ to an unnamed temporary may be used [-Werror=dangling-pointer=]
  662 |         err = pacs_gatt_notify(conn, uuid, pacs_svc.attrs, &location_le,
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  663 |                                   sizeof(location_le));
      |                                   ~~~~~~~~~~~~~~~~~~~~
In file included from zephyr/include/zephyr/bluetooth/gatt.h:28,
                 from zephyr/subsys/bluetooth/audio/pacs.c:21:
zephyr/include/zephyr/bluetooth/uuid.h:113:52: note: unnamed temporary defined here
  113 |         ((struct bt_uuid *) ((struct bt_uuid_16[]) {BT_UUID_INIT_16(value)}))
      |                                                    ^
zephyr/include/zephyr/bluetooth/uuid.h:4750:9: note: in expansion of macro ‘BT_UUID_DECLARE_16’
 4750 |         BT_UUID_DECLARE_16(BT_UUID_PACS_SNK_LOC_VAL)
      |         ^~~~~~~~~~~~~~~~~~
zephyr/subsys/bluetooth/audio/pacs.c:645:24: note: in expansion of macro ‘BT_UUID_PACS_SNK_LOC’
  645 |                 uuid = BT_UUID_PACS_SNK_LOC;
      |                        ^~~~~~~~~~~~~~~~~~~~
zephyr/subsys/bluetooth/audio/pacs.c: In function ‘pac_notify’:
zephyr/subsys/bluetooth/audio/pacs.c:702:15: error: dangling pointer ‘uuid’ to an unnamed temporary may be used [-Werror=dangling-pointer=]
  702 |         err = pacs_gatt_notify(conn, uuid, pacs_svc.attrs,
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  703 |                                read_buf.data, read_buf.len);
      |                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
zephyr/include/zephyr/bluetooth/uuid.h:113:52: note: unnamed temporary defined here
  113 |         ((struct bt_uuid *) ((struct bt_uuid_16[]) {BT_UUID_INIT_16(value)}))
      |                                                    ^
zephyr/include/zephyr/bluetooth/uuid.h:4741:9: note: in expansion of macro ‘BT_UUID_DECLARE_16’
 4741 |         BT_UUID_DECLARE_16(BT_UUID_PACS_SNK_VAL)
      |         ^~~~~~~~~~~~~~~~~~
zephyr/subsys/bluetooth/audio/pacs.c:681:24: note: in expansion of macro ‘BT_UUID_PACS_SNK’
  681 |                 uuid = BT_UUID_PACS_SNK;
      |                        ^~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

**Environment (please complete the following information):**
 - OS: Ubuntu
 - Toolchain: gcc 12 (gcc (Ubuntu 12.3.0-1ubuntu1~22.04) 12.3.0)
 - Commit SHA: Todays main : 5e01466d977f04c28fb9f0adf872623f82ae3f51

**Notes**
gcc 11 does not warn about this issue

--- a/subsys/bluetooth/audio/pacs.c
+++ b/subsys/bluetooth/audio/pacs.c
@@ -308,6 +308,8 @@ static ssize_t snk_read(struct bt_conn *conn, const struct bt_gatt_attr *attr,
 }
 
 #if defined(CONFIG_BT_PAC_SNK_NOTIFIABLE)
+static const struct bt_uuid *pacs_snk_uuid = BT_UUID_PACS_SNK;
+
 static void snk_cfg_changed(const struct bt_gatt_attr *attr, uint16_t value)
 {
 	LOG_DBG("attr %p value 0x%04x", attr, value);
@@ -351,6 +353,8 @@ static ssize_t snk_loc_read(struct bt_conn *conn,
 }
 
 #if defined(CONFIG_BT_PAC_SNK_LOC_NOTIFIABLE)
+static const struct bt_uuid *pacs_snk_loc_uuid = BT_UUID_PACS_SNK_LOC;
+
 static void snk_loc_cfg_changed(const struct bt_gatt_attr *attr, uint16_t value)
 {
 	LOG_DBG("attr %p value 0x%04x", attr, value);
@@ -430,12 +434,14 @@ static ssize_t src_read(struct bt_conn *conn, const struct bt_gatt_attr *attr,
 	return ret_val;
 }
 
-#if defined(CONFIG_BT_PAC_SRC_LOC_NOTIFIABLE)
+#if defined(CONFIG_BT_PAC_SRC_NOTIFIABLE)
+static const struct bt_uuid *pacs_src_uuid = BT_UUID_PACS_SRC;
+
 static void src_cfg_changed(const struct bt_gatt_attr *attr, uint16_t value)
 {
 	LOG_DBG("attr %p value 0x%04x", attr, value);
 }
-#endif /* CONFIG_BT_PAC_SRC_LOC_NOTIFIABLE */
+#endif /* CONFIG_BT_PAC_SRC_NOTIFIABLE */
 
 static inline int set_src_available_contexts(uint16_t contexts)
 {
@@ -474,6 +480,8 @@ static ssize_t src_loc_read(struct bt_conn *conn,
 }
 
 #if defined(CONFIG_BT_PAC_SRC_LOC_NOTIFIABLE)
+static const struct bt_uuid *pacs_src_loc_uuid = BT_UUID_PACS_SRC_LOC;
+
 static void src_loc_cfg_changed(const struct bt_gatt_attr *attr, uint16_t value)
 {
 	LOG_DBG("attr %p value 0x%04x", attr, value);
@@ -552,15 +560,17 @@ static sys_slist_t *pacs_get(enum bt_audio_dir dir)
 		      BT_PACS_SNK_PROP, \
 		      BT_GATT_PERM_READ_ENCRYPT, \
 		      _read, NULL, NULL), \
-	IF_ENABLED(CONFIG_BT_PAC_SNK_LOC_NOTIFIABLE, (BT_AUDIO_CCC(snk_cfg_changed),))
+	IF_ENABLED(CONFIG_BT_PAC_SNK_NOTIFIABLE, (BT_AUDIO_CCC(snk_cfg_changed),))
 
 #define BT_PACS_SNK_LOC_PROP \
 	BT_GATT_CHRC_READ \
 	IF_ENABLED(CONFIG_BT_PAC_SNK_LOC_WRITEABLE, (|BT_GATT_CHRC_WRITE)) \
 	IF_ENABLED(CONFIG_BT_PAC_SNK_LOC_NOTIFIABLE, (|BT_GATT_CHRC_NOTIFY))
+
 #define BT_PACS_SNK_LOC_PERM \
 	BT_GATT_PERM_READ_ENCRYPT \
 	IF_ENABLED(CONFIG_BT_PAC_SNK_LOC_WRITEABLE, (|BT_GATT_PERM_WRITE_ENCRYPT))
+
 #define BT_PACS_SNK_LOC(_read) \
 	BT_AUDIO_CHRC(BT_UUID_PACS_SNK_LOC, \
 		      BT_PACS_SNK_LOC_PROP, \
@@ -578,15 +588,17 @@ static sys_slist_t *pacs_get(enum bt_audio_dir dir)
 		      BT_PACS_SRC_PROP, \
 		      BT_GATT_PERM_READ_ENCRYPT, \
 		      _read, NULL, NULL), \
-	IF_ENABLED(CONFIG_BT_PAC_SRC_LOC_NOTIFIABLE, (BT_AUDIO_CCC(src_cfg_changed),))
+	IF_ENABLED(CONFIG_BT_PAC_SRC_NOTIFIABLE, (BT_AUDIO_CCC(src_cfg_changed),))
 
 #define BT_PACS_SRC_LOC_PROP \
 	BT_GATT_CHRC_READ \
 	IF_ENABLED(CONFIG_BT_PAC_SRC_LOC_WRITEABLE, (|BT_GATT_CHRC_WRITE)) \
 	IF_ENABLED(CONFIG_BT_PAC_SRC_LOC_NOTIFIABLE, (|BT_GATT_CHRC_NOTIFY))
+
 #define BT_PACS_SRC_LOC_PERM \
 	BT_GATT_PERM_READ_ENCRYPT \
 	IF_ENABLED(CONFIG_BT_PAC_SRC_LOC_WRITEABLE, (|BT_GATT_PERM_WRITE_ENCRYPT))
+
 #define BT_PACS_SRC_LOC(_read) \
 	BT_AUDIO_CHRC(BT_UUID_PACS_SRC_LOC, \
 		      BT_PACS_SRC_LOC_PROP, \
@@ -606,6 +618,7 @@ static sys_slist_t *pacs_get(enum bt_audio_dir dir)
 #define BT_PACS_SUPPORTED_CONTEXT_PROP \
 	BT_GATT_CHRC_READ \
 	IF_ENABLED(CONFIG_BT_PACS_SUPPORTED_CONTEXT_NOTIFIABLE, (|BT_GATT_CHRC_NOTIFY))
+
 #define BT_PAC_SUPPORTED_CONTEXT(_read) \
 	BT_AUDIO_CHRC(BT_UUID_PACS_SUPPORTED_CONTEXT, \
 		      BT_PACS_SUPPORTED_CONTEXT_PROP, \
@@ -636,13 +649,13 @@ static int pac_notify_loc(struct bt_conn *conn, enum bt_audio_dir dir)
 {
 	uint32_t location_le;
 	int err;
-	struct bt_uuid *uuid;
+	const struct bt_uuid *uuid;
 
 	switch (dir) {
 	case BT_AUDIO_DIR_SINK:
 #if defined(CONFIG_BT_PAC_SNK_LOC_NOTIFIABLE)
 		location_le = sys_cpu_to_le32(pacs_snk_location);
-		uuid = BT_UUID_PACS_SNK_LOC;
+		uuid = pacs_snk_loc_uuid;
 		break;
 #else
 		return -ENOTSUP;
@@ -650,7 +663,7 @@ static int pac_notify_loc(struct bt_conn *conn, enum bt_audio_dir dir)
 	case BT_AUDIO_DIR_SOURCE:
 #if defined(CONFIG_BT_PAC_SRC_LOC_NOTIFIABLE)
 		location_le = sys_cpu_to_le32(pacs_src_location);
-		uuid = BT_UUID_PACS_SRC_LOC;
+		uuid = pacs_src_loc_uuid;
 		break;
 #else
 		return -ENOTSUP;
@@ -659,8 +672,7 @@ static int pac_notify_loc(struct bt_conn *conn, enum bt_audio_dir dir)
 		return -EINVAL;
 	}
 
-	err = pacs_gatt_notify(conn, uuid, pacs_svc.attrs, &location_le,
-				  sizeof(location_le));
+	err = pacs_gatt_notify(conn, uuid, pacs_svc.attrs, &location_le, sizeof(location_le));
 	if (err != 0 && err != -ENOTCONN) {
 		LOG_WRN("PACS notify_loc failed: %d", err);
 		return err;
@@ -673,17 +685,19 @@ static int pac_notify(struct bt_conn *conn, enum bt_audio_dir dir)
 {
 	int err = 0;
 	sys_slist_t *pacs;
-	struct bt_uuid *uuid;
+	const struct bt_uuid *uuid;
 
 	switch (dir) {
+#if defined(CONFIG_BT_PAC_SNK_NOTIFIABLE)
 	case BT_AUDIO_DIR_SINK:
-		__ASSERT(IS_ENABLED(CONFIG_BT_PAC_SNK_NOTIFIABLE), "Sink PAC not notifiable.\n");
-		uuid = BT_UUID_PACS_SNK;
+		uuid = pacs_snk_uuid;
 		break;
+#endif /* CONFIG_BT_PAC_SNK_NOTIFIABLE */
+#if defined(CONFIG_BT_PAC_SRC_NOTIFIABLE)
 	case BT_AUDIO_DIR_SOURCE:
-		__ASSERT(IS_ENABLED(CONFIG_BT_PAC_SRC_NOTIFIABLE), "Source PAC not notifiable.\n");
-		uuid = BT_UUID_PACS_SRC;
+		uuid = pacs_src_uuid;
 		break;
+#endif /* CONFIG_BT_PAC_SRC_NOTIFIABLE */
 	default:
 		return -EINVAL;
 	}

